### PR TITLE
Use Calypso's eslint rules and require no errors for build.

### PR DESCRIPTION
**Pull request by dereksmart**
Use Calypso's eslint rules and require no errors for build.  

This PR seeks to change the eslint rules and require no errors on build/watch.  This PR also includes fixes for all `errors`, as there are still over 1,000 warnings we'll need to address later.  

To Test: 
- First `npm distclean && cache clean`
- `yarn build && yarn watch`: Build/watch plugin, make sure no errors
- `gulp eslint` to view warnings

Changes in this PR: 
- Now using calypso eslint rules verbatim, with a couple of differences.
    - Ignoring CSS className rule `"wpcalypso/jsx-classname-namespace"`.  The syntax requires the className prefix to match the directory name.  We namespace all of our classes `jp-`, but our directory names are not, so the rule was failing us.  I don't think this is very important to match and is not worth refactoring a dozen or so directory names to pass. 
    - Ignoring the `"max-len"` rule of 140 characters.  We have over 200 instances of this in the plugin, and although we should not condone this, I simply don't feel it's worth fixing all instances of this.  In the future, please try and be cognizant of this and keep your lines short! 
    - `"react/no-danger"`: I set this to be a warning rather than an error.  I did this because we're using it in a few key places ATM, and I wasn't up for the refactor now, but I think it's something we should find a solution to in the future.  
    - `"lodash/import-scope": [ 2, "method" ]`: Added rule for importing lodash functions.  We were running into issues where `_` would get overwritten when importing the entire lodash library rather than just the function.  This will now `error` if you're importing a function incorrectly.  
```
import map from 'lodash/map' //correct
import {map} from 'lodash' //incorrect
```
- Updated/added node modules: 
    - Updated "eslint": "3.8.1" (also moved to dev-dependencies)
    - Updated "babel-eslint": "6.1.2"
    - Updated "eslint-plugin-react": "6.4.1"(also moved to dev-dependencies)
    - Added "gulp-eslint": "3.0.1"
    - Added "eslint-config-wpcalypso": "0.6.0"
    - Added "eslint-plugin-lodash": "2.3.4"
- New Gulp task `gulp eslint`
    - Will run the eslint rules on all js/jsx files in `_inc/client/`, excluding any `test` files. 
- Builds will now fail if `errors` on found.
- `watch` tasks will display `errors` if found during watch.

### CODE REVIEW

File: _inc/client/components/forms/index.jsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const FormFieldset = React.createClass( {
 								<Gridicon icon="help-outline" />
 								<span className="screen-reader-text">{ __( 'Learn More' ) }</span>
 							</Button>
-						  </div>
+					</div>
```
Comment (oskosk) [inline _inc/client/components/forms/index.jsx]: Indentation weird here :P

Comment (dereksmart) [inline _inc/client/components/forms/index.jsx]: Better than mixed tabs/spaces lol

**Comment by oskosk** [comment]
LGTM. No linting errors shown with testing instructions. Lots of warning due to new eslint rules. :shipit:

**Comment by eliorivero** [comment]
Nice! Works fine.
Loving that it now shows incorrect lodash imports as errors
```
 error    Do not import from the full Lodash module              lodash/import-scope
```
and also shows them in phpStorm 👍 
captura de pantalla 2017-02-10 a las 13 27 31

**Pull request merged**